src/Transformations.jl

```diff
@@ -0,0 +1,180 @@
+module Transformations
+
+using Statistics
+
+using Oceananigans.Fields: interior
+using Oceananigans.OutputReaders: FieldTimeSeries
+
+#####
+##### Identity normalization
+#####
+
+struct Transformation{T, X, N}
+    time :: T
+    space :: X
+    normalization :: N
+end
+
+"""
+    Transformation(; time=nothing, space=nothing, normalization=nothing)
+
+"""
+Transformation(; time=nothing, space=nothing, normalization=nothing) =
+    Transformation(time, space, normalization)
+
+"""
+    compute_time_transformation(user_time_transformation, fts)
+
+Compute a time transformation for the field time series `fts`
+given `user_time_transformation`.
+
+By default, we include all time slices except the initial condition.
+"""
+function compute_time_transformation(::Nothing, fts)
+    Nt = length(fts.times)
+    return 2:Nt
+end
+
+compute_space_transformation(::Nothing, fts) = nothing
+compute_normalization(::Nothing, transformation, fts) = nothing
+
+function compute_transformation(transformation, field_time_series)
+    time = compute_time_transformation(transformation.time, field_time_series)
+    space = compute_space_transformation(transformation.space, field_time_series)
+    normalization = compute_normalization(transformation.normalization, transformation, field_time_series)
+    return Transformation(time, space, normalization)
+end
+
+#####
+##### Time transformations
+#####
+
+time_transform(::Nothing, data) = data
+time_transform(indices::UnitRange, data) = data[:, :, :, indices]
+
+function time_transform(weights::AbstractVector, data)
+    weights = reshape(weights, 1, 1, 1, length(weights))
+    return data .* weights
+end
+
+#####
+##### Space transformations
+#####
+
+space_transform(::Nothing, data) = data
+space_transform(indices::UnitRange, data) = data[indices[1], indices[2], indices[3], :]
```

Comment:
```suggestion
space_transform(indices::Tuple{UnitRange, UnitRange, UnitRange}, data) = data[indices[1], indices[2], indices[3], :]
```

??

Reply:
I think actually we just want `Tuple` here (basically anything that can index is valid)

Reply:
And good catch, `UnitRange` is not right...